src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -371,55 +364,79 @@ def check_kube_connection(configuration):
-                              "Learn more at https://aka.ms/arc/k8s/onboarding-helm-install")
-    return output_helm_version.decode('ascii')
+def install_helm_client():
+    # Return helm client path set by user
+    if os.getenv('HELM_CLIENT_PATH'):
+        return os.getenv('HELM_CLIENT_PATH')
+
+    # Fetch system related info
+    operating_system = platform.system().lower()
+    machine_type = platform.machine()
+
+    # Send machine telemetry
+    telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.MachineType': machine_type})
+
+    # Set helm binary download & install locations
+    if(operating_system == 'windows'):
+        download_location_string = f'.azure\\helm\\{consts.HELM_VERSION}\\helm-{consts.HELM_VERSION}-{operating_system}-amd64.zip'
+        install_location_string = f'.azure\\helm\\{consts.HELM_VERSION}\\{operating_system}-amd64\\helm.exe'
+        requestUri = f'{consts.HELM_STORAGE_URL}/helm/helm-{consts.HELM_VERSION}-{operating_system}-amd64.zip'
+    elif(operating_system == 'linux' or operating_system == 'darwin'):
+        download_location_string = f'.azure/helm/{consts.HELM_VERSION}/helm-{consts.HELM_VERSION}-{operating_system}-amd64.tar.gz'
+        install_location_string = f'.azure/helm/{consts.HELM_VERSION}/{operating_system}-amd64/helm'
+        requestUri = f'{consts.HELM_STORAGE_URL}/helm/helm-{consts.HELM_VERSION}-{operating_system}-amd64.tar.gz'
+    else:
+        telemetry.set_exception(exception='Unsupported OS for installing helm client', fault_type=consts.Helm_Unsupported_OS_Fault_Type,
+                                summary=f'{operating_system} is not supported for installing helm client')
+        raise ClientRequestError(f'The {operating_system} platform is not currently supported for installing helm client.')
+
+    download_location = os.path.expanduser(os.path.join('~', download_location_string))
+    download_dir = os.path.dirname(download_location)
+    install_location = os.path.expanduser(os.path.join('~', install_location_string))
+
+    # Download compressed halm binary if not already present
+    if not os.path.isfile(download_location):
+        # Downloading compressed helm client executable
+        print("Downloading helm client for first time. This can take few minutes...")
+        try:
+            response = urllib.request.urlopen(requestUri)
+        except Exception as e:
+            telemetry.set_exception(exception=e, fault_type=consts.Download_Helm_Fault_Type,
+                                    summary='Unable to download helm client.')
+            raise CLIInternalError("Failed to download helm client.", recommendation="Please check your internet connection." + str(e))
+
+        responseContent = response.read()
+        response.close()
+
+        # Creating the helm folder if it doesnt exist
+        if not os.path.exists(download_dir):
+            try:
```

Comment:
This check should be moved before downloading binary.

Reply:
done